Assets/Code/Components/AI/AIController.cs

```diff
@@ -18,7 +49,47 @@ private void Start()
 
         private void Update()
         {
+            CheckIfPlayerIsInSight();
             tree.Update();
+            Debug.DrawRay(this.transform.position + Vector3.up * eyesRelativePosition, Quaternion.Euler(0, -fieldOfVision, 0) * this.transform.forward * this.sightDistance, Color.red);
+            Debug.DrawRay(this.transform.position + Vector3.up * eyesRelativePosition, Quaternion.Euler(0, fieldOfVision, 0) * this.transform.forward * this.sightDistance, Color.red);
         }
+
+        /// <summary>
+        /// Checks if a player is on sight, and updates the property 'PlayerOnSight'. In the future the BT should store the state variables that can affect the behviour.
+        /// </summary>
+        public void CheckIfPlayerIsInSight()
+        {
+            foreach (var player in worldData.Players)
+            {
+                var eyesPosition = this.EyesPosition;
+                var vectorBetween = player.GetComponent<CapsuleCollider>().bounds.center - eyesPosition;
```

Comment:
var lo que hace es que infiere el tipo según lo que le ponga a la derecha del "=". Si no puede inferirlo en compilación, entonces tira error. No tiene costos de performance ni nada. Sólo a veces puede dejar más claro lo que es cada cosa.

Ej:

var eyesPosition = this.EyesPosition;  // El compilador sabe que EyesPosition es un Transform, por lo tanto sabe que "eyesPosition" es de tipo transform.

Otro ejemplo:

var asd = null; // Error, porque de null no se puede inferir el tipo. Ahora te toca a ti!